atspi-connection/src/p2p.rs

```diff
@@ -0,0 +1,768 @@
+	pub(crate) fn spawn_peer_listener_task(&self, conn: &zbus::Connection) {
+		// Clone the `Peers` and `Connection` to move them into the async task.
+		// This is necessary because the async task needs to own these values.
+		let peers = self.clone();
+		let conn = conn.clone();
+		let dbus_proxy = futures_lite::future::block_on(DBusProxy::new(&conn))
+			.expect("Failed to create DBusProxy");
+
+		let executor = conn.executor().clone();
+
+		executor.spawn(async move {
+			let Ok(mut name_owner_changed_stream) = dbus_proxy.receive_name_owner_changed().await.inspect_err(|#[allow(unused_variables)] err| {
+				#[cfg(feature = "tracing")]
+				debug!("Failed to receive `NameOwnerChanged` stream: {err}");
+			}) else {
+				return;
+			};
+
+			while let Some(name_owner_event) = name_owner_changed_stream.next().await {
+					let Ok(args) = name_owner_event.args() else {
+						#[cfg(feature = "tracing")]
+						tracing::debug!("Received name owner changed event without args, skipping.");
+						continue;
+					};
+					let name = args.name().clone();
+					let new = args.new_owner().clone();
+					let old = args.old_owner().clone();
+
+					// `NameOwnerChanged` table (U = Unique, W = Well-Known):
+					// | Name | Old Owner | New Owner | Operation |
+					// |------|-----------|-----------|----------|
+					// |   U  |   None    |  Some(U)  |  Add     |
+					// |   U  |   Some(U) |    None   |  Remove  |
+					// |   W  |   None    |  Some(U)  |  Add     |
+					// |   W  |   Some(U) |    None   |  Remove  |
+					// |   W  |   Some(U) |  Some(U)  |  Replace |
+
+					match name {
+						BusName::Unique(unique_name) => {
+							// `zvariant:Optional` has deref target `Option`.
+							match (&*old, &*new) {
+								// Application appeared on the bus.
+								(None, Some(new_owner)) => {
+									debug_assert_eq!(new_owner, &unique_name, "When a name appears on the bus, the new owner must be the unique name itself.");
+
+									if let Ok(()) = peers.insert_unique(&unique_name, &conn).await.inspect_err(|#[allow(unused_variables)] err| {
+										#[cfg(feature = "tracing")]
+										warn!("Failed to insert unique name: {unique_name}: {err}");
+									}) {
+										#[cfg(feature = "tracing")]
+										info!("Inserted unique name: {unique_name} into the peer list.");
+									};
+
+								}
+								// Unique name left the bus.
+								(Some(old), None) => {
+									debug_assert!(old == &unique_name, "When a unique name is removed from the bus, the old owner must be the unique name itself.");
+									peers.remove_unique(&unique_name).await;
+
+									#[cfg(feature = "tracing")]
+									info!("Peer with unique name: {unique_name} left the bus - removed from peer list.");
+								}
+
+								// Unknown combination.
+								(_, _) => {
+									#[cfg(feature = "tracing")]
+									debug!("NameOwnerChanged` with unique name: {unique_name} has unknown argument combination ({old:?}, {new:?}).");
```

Comment:
Some of these are `debug!` and some are `debug_assert!`; how do you differentiate the two cases?